boot/bootstate20.go

```diff
@@ -769,3 +769,32 @@ func (brs20 *bootState20RecoverySystem) markSuccessful(update bootStateUpdate) (
 func recoverySystemsBootState(dev Device) *bootState20RecoverySystem {
 	return &bootState20RecoverySystem{dev: dev}
 }
+
+// bootState20Model implements the successfulBootState interface for device
+// model related bookkeeping
+type bootState20Model struct {
+	dev Device
+}
+
+func (brs20 *bootState20Model) markSuccessful(update bootStateUpdate) (bootStateUpdate, error) {
+	u20, err := toBootStateUpdate20(update)
+	if err != nil {
+		return nil, err
+	}
+
+	// sign key ID was not being populated in earlier versions of snapd, try
+	// to remedy that
+	if u20.writeModeenv.SignKeyID == "" {
+		newM, err := u20.writeModeenv.Copy()
```

Comment:
why the copy? writeModeenv is already a copy itself. Afaict the rest of code just writes into writeModeenv

Reply:
i've dropped the copy